SudokuSolver/SudokuSolver/CrossHatching/SudokuSolver.cs

```diff
@@ -0,0 +1,134 @@
+﻿using Microsoft.Toolkit.Extensions;
+using SudokuSolver.Common;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SudokuSolver.CrossHatching
+{
+    class SudokuSolver : BaseSudokuSolver
+    {
+        private Dictionary<int, List<(int i, int j)>> filledPos;
+        private Dictionary<int, int> remaining;
+        private Dictionary<int, Dictionary<int, List<int>>> graph;
+
+        public SudokuSolver() { }
+        public override Result<Sudoku> Solve(Sudoku sudoku)
+        {
+            BuildFilledPosAndRemaining(sudoku);
+            BuildGraph(sudoku);
+
+            remaining = remaining.OrderBy(kvpair => kvpair.Value).ToDictionary(x => x.Key, x => x.Value);
+
+            var remainingKeys = remaining.Keys.ToList();
+            int currentRemKey = 0;
+            int currentRow = 0;
+
+            Stopwatch sw = new Stopwatch();
```

Comment:
to jest dobrze, ale jakbyś chciała to z tego co kojarzę jest metoda statyczna która robi ci nowy, już wystartowany StopWatch ale to po prostu taki "quality of life"